kani-driver/src/call_cbmc.rs

```diff
@@ -164,6 +184,94 @@ impl KaniSession {
+            if let Some(result_idx) = result_idx {
+                let result = items.remove(result_idx);
+                if let ParserItem::Result { result } = result {
+                    let number_failed_properties =
+                        result.iter().filter(|prop| prop.status == CheckStatus::Failure).count();
+                    let status = if number_failed_properties == 0 {
+                        VerificationStatus::Success
+                    } else {
+                        VerificationStatus::Failure
+                    };
+                    VerificationResult {
+                        status,
+                        messages: Some(items),
+                        results: Some(result),
+                        exit_status: output.process_status,
+                        runtime: start_time.elapsed(),
+                    }
+                } else {
+                    unreachable!() // We filtered for this to be true
+                }
+            } else {
+                // We never got results from CBMC - also shouldn't really happen unless something went wrong
+                VerificationResult {
+                    status: VerificationStatus::Failure,
+                    messages: Some(items),
+                    results: None,
+                    exit_status: output.process_status,
+                    runtime: start_time.elapsed(),
+                }
+            }
+        } else {
+            // We don't have ANY messages to report - shouldn't really happen unless something went wrong
+            VerificationResult {
+                status: VerificationStatus::Failure,
+                messages: None,
+                results: None,
+                exit_status: output.process_status,
+                runtime: start_time.elapsed(),
+            }
+        }
+    }
+    pub fn mock_success() -> VerificationResult {
+        VerificationResult {
+            status: VerificationStatus::Success,
+            messages: None,
+            results: None,
+            exit_status: 42,
+            runtime: Duration::from_secs(0),
+        }
+    }
+    fn mock_failure() -> VerificationResult {
+        VerificationResult {
+            status: VerificationStatus::Failure,
+            messages: None,
+            results: None,
+            exit_status: 42,
+            runtime: Duration::from_secs(0),
+        }
+    }
+
+    pub fn render(&self, output_format: &OutputFormat) -> String {
+        if let Some(results) = &self.results {
+            let show_checks = matches!(output_format, OutputFormat::Regular);
+            let mut result = format_result(results, show_checks);
+            writeln!(result, "Verification Time: {}s", self.runtime.as_secs_f32()).unwrap();
+            result
+        } else {
+            let verification_result = console::style("FAILED").red();
+            format!(
+                "\nCBMC failed with status {}\nVERIFICATION:- {}\n",
```

Comment:
I don't know if it makes sense to provide a verification result in this case... We could say something like "verification couldn't be performed".

Reply:
I thought it best to keep a very simple/consistent response for this. The verification did fail, the reasons why will be above that line, just like normal output.